source/extensions/filters/network/dubbo_proxy/router/router_impl.cc

```diff
@@ -0,0 +1,342 @@
+        Tcp::ConnectionPool::PoolFailureReason::RemoteConnectionFailure);
+    break;
+  case Network::ConnectionEvent::LocalClose:
+    upstream_request_->onResetStream(
+        Tcp::ConnectionPool::PoolFailureReason::LocalConnectionFailure);
+    break;
+  default:
+    // Connected is consumed by the connection pool.
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+}
+
+const Network::Connection* Router::downstreamConnection() const {
+  return callbacks_ != nullptr ? callbacks_->connection() : nullptr;
+}
+
+void Router::cleanup() {
+  if (upstream_request_) {
+    upstream_request_.reset();
+  }
+}
+
+Router::UpstreamRequest::UpstreamRequest(Router& parent, Tcp::ConnectionPool::Instance& pool,
+                                         MessageMetadataSharedPtr& metadata,
+                                         SerializationType serialization_type,
+                                         ProtocolType protocol_type)
+    : parent_(parent), conn_pool_(pool), metadata_(metadata),
+      deserializer_(
+          NamedDeserializerConfigFactory::getFactory(serialization_type).createDeserializer()),
+      protocol_(NamedProtocolConfigFactory::getFactory(protocol_type).createProtocol()),
+      request_complete_(false), response_started_(false), response_complete_(false) {}
+
+Router::UpstreamRequest::~UpstreamRequest() {}
+
+Network::FilterStatus Router::UpstreamRequest::start() {
+  Tcp::ConnectionPool::Cancellable* handle = conn_pool_.newConnection(*this);
+  if (handle) {
+    // Pause while we wait for a connection.
+    conn_pool_handle_ = handle;
+    return Network::FilterStatus::StopIteration;
+  }
+
+  return Network::FilterStatus::Continue;
+}
+
+void Router::UpstreamRequest::resetStream() {
+  if (conn_pool_handle_) {
+    ASSERT(!conn_data_);
+    conn_pool_handle_->cancel(Tcp::ConnectionPool::CancelPolicy::Default);
+    conn_pool_handle_ = nullptr;
+  }
+
+  if (conn_data_) {
+    ASSERT(!conn_pool_handle_);
+    conn_data_->connection().close(Network::ConnectionCloseType::NoFlush);
+    conn_data_.reset();
+  }
+}
+
+void Router::UpstreamRequest::encodeData(Buffer::Instance& data) {
+  if (!conn_data_) {
+    ENVOY_STREAM_LOG(trace, "buffering {} bytes", *parent_.callbacks_, data.length());
+    if (!buffered_request_body_) {
+      buffered_request_body_ = std::make_unique<Envoy::Buffer::OwnedImpl>();
+    }
+
+    buffered_request_body_->move(data);
+  } else {
+    ENVOY_STREAM_LOG(trace, "proxying {} bytes", *parent_.callbacks_, data.length());
+    conn_data_->connection().write(data, false);
+  }
+}
+
+void Router::UpstreamRequest::onPoolFailure(Tcp::ConnectionPool::PoolFailureReason reason,
+                                            Upstream::HostDescriptionConstSharedPtr host) {
+  ENVOY_LOG(warn, "dubbo upstream request: connection failure '{}'", host->address()->asString());
+
+  conn_pool_handle_ = nullptr;
+
+  // Mimic an upstream reset.
+  onUpstreamHostSelected(host);
+  onResetStream(reason);
+
```

Comment:
How to continue to call the next filter callback?

Reply:
I've added the continue decoding logic.